core/src/main/scala/org/apache/spark/api/r/RBackendHandler.scala

```diff
@@ -224,7 +224,8 @@ private[r] class RBackendHandler(server: RBackend)
                 case _ => parameterType
               }
             }
-            if (!parameterWrapperType.isInstance(args(i))) {
+            if (!(!parameterType.isPrimitive && args(i) == null) &&
+                !parameterWrapperType.isInstance(args(i))) {
```

Comment:
OK. done.

Anyway, I feel painful that Scala does not support 'continue' statement. So the code here still not so good.